cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/SourceControlManagementHttpHandler.java

```diff
@@ -46,48 +51,50 @@
 @Path(Constants.Gateway.API_VERSION_3 + "/namespaces/{namespace-id}/repository")
 public class SourceControlManagementHttpHandler extends AbstractAppFabricHttpHandler {
   private final SourceControlManagementService sourceControlService;
+  private final FeatureFlagsProvider featureFlagsProvider;
   private static final Gson GSON = new Gson();
 
   @Inject
-  SourceControlManagementHttpHandler(SourceControlManagementService sourceControlService) {
+  SourceControlManagementHttpHandler(CConfiguration configuration,
+                                     SourceControlManagementService sourceControlService) {
     this.sourceControlService = sourceControlService;
+    this.featureFlagsProvider = new DefaultFeatureFlagsProvider(configuration);
   }
 
   @PUT
   @Path("/")
   @AuditPolicy(AuditDetail.REQUEST_BODY)
   public void setRepository(FullHttpRequest request, HttpResponder responder,
                             @PathParam("namespace-id") String namespaceId) throws Exception {
+    checkSourceControlFeatureFlag();
     NamespaceId namespace = validateNamespaceId(namespaceId);
     RepositoryConfigRequest repoRequest = getRepositoryConfigRequest(request);
 
-    if (repoRequest == null || repoRequest.getConfig() == null) {
-      responder.sendJson(HttpResponseStatus.BAD_REQUEST,
-                         GSON.toJson(new SetRepositoryResponse("Repository configuration must be specified.")));
-    }
-
     try {
+      if (repoRequest == null || repoRequest.getConfig() == null) {
+        throw new InvalidRepositoryConfigException("Repository configuration must be specified.");
```

Comment:
Moving these lines into the try block and catch `InvalidRepositoryConfigException` for consistency, also avoid missing returns after the responder sends back